packages/core/contracts/optimistic-asserter/implementation/OptimisticAsserter.sol

```diff
@@ -82,6 +102,24 @@ contract OptimisticAsserter is OptimisticAsserterInterface, Lockable, Ownable, M
             );
     }
 
+    /**
+     * @notice Asserts a truth about the world, using a fully custom configuration.
+     * @dev The caller must approve this contract to spend at least bond amount of currency.
+     * @param claim the truth claim being asserted. This is an assertion about the world, and is verified by disputers.
+     * @param asserter receives bonds back at settlement. This could be msg.sender (if sent from an EOA) or a contract
+     * address or any other account that the caller wants to receive the bond at settlement time.
+     * @param callbackRecipient if configured, this address will receive a function call assertionResolvedCallback and
+     * assertionDisputedCallback at resolution or dispute respectively. Enables dynamic responses to these events.
+     * @param escalationManager if configured, this address will control escalation properties of the assertion. This
+     * this means a) choosing to arbitrate via the UMA DVM, b) choosing to discard assertions on dispute, or choosing to
+     * validate disputes. Combining these, the asserter can define their own security properties the assertion.
+     * @param currency bond currency pulled from the caller and held in escrow until the assertion is resolved.
+     * @param bond amount of currency to pull from the caller and hold in escrow until the assertion is resolved.
+     * @param liveness time to wait before the assertion can be resolved. Assertion can be disputed in this time.
+     * @param identifier UMA DVM identifier to use for price requests in the event of a dispute.
+     * @param domainId identifier provided by the asserter to enable additional grouping between assertions within a
+     * known "domain". Not used in this contract but can be used by escalationManager to associate sets of assertions.
```

Comment:
```suggestion
     * @param domainId optional domain that can be used to relate this assertion to other assertions in the escalationManager.
     * This can be used by the configured escalationManager to define custom behavior for groups of assertions.
     * This is typically used for "escalation games" by changing bonds or other assertion properties
     * based on the other assertions that have come before. If no escalationManager is configured or a domain is not needed,
     * this value should be set to bytes32(0) to reduce gas costs.
```